src/RA_Achievement.h

```diff
@@ -4,7 +4,18 @@
 
 #include "RA_Condition.h"
 
+namespace ra {
+namespace detail {
 
+enum class AchievementSetType
+{
+    Core,
+    Unofficial,
+    Local
+};
+
+} /* namespace detail */
+} /* namespace ra */
```

Comment:
This couldn't be a member type of AchievementSet because it made a circular dependency resulting in `AchievementSet  `and `Achievement `being incomplete types. I just aliased it instead. They are type aliases so we really shouldn't get any type clashes. Most of the time we used the AchievementSet::Type one like discussed before.

It's in a detail namespace to prevent it from being used directly as the member type was supposed be the one used.

Reply:
I believe you can eliminate the circular dependency by removing the `Achievement::IsCoreAchievement` method and removing the `SetActive(IsCoreAchievement())` call from `Achievement::Parse` and just call `SetActive(TRUE)` after calling `Parse` in `AchievementSet::LoadFromFile` here:

```
                    if (nFlags == 3 && m_nSetType == Core)
                    {
                        Achievement& newAch = AddAchievement();
                        newAch.Parse(AchievementsData[i]);
                        newAch.SetActive(TRUE); //Activate core by default
                    }
```

I'd rather clean up the problem than work around it.

Reply:
So should we just call it "Achievement::Type" and/or alias it in achievementset?